csharp/BYOAuthFlow/Program.cs

```diff
@@ -0,0 +1,225 @@
+        private static string RefreshToken =>
+            Environment.GetEnvironmentVariable("IMPROBABLE_REFRESH_TOKEN") ?? "5b6c50ed-f488-45f2-b02b-e6caa346b26e";
+
+        /// <summary>
+        ///     This contains the implementation of the "bring your own auth flow" scenario.
+        ///     1. Start a cloud deployment.
+        ///     2. Generate a Player Identity Token.
+        ///     3. List deployments.
+        ///     4. Generate a Login Token for a selected deployment.
+        ///     5. Connect to the deployment using the Player Identity Token and the Login Token.
+        /// </summary>
+        /// <param name="args"></param>
+        private static void Main(string[] args)
+        {
+            Console.WriteLine("Starting Cloud Scenario");
+            var platformCredentials = new PlatformRefreshTokenCredential(RefreshToken);
+            RunScenario(
+                "locator.improbable.io",
+                PlayerIdentityTokenServiceClient.Create(credentials: platformCredentials),
+                LoginTokenServiceClient.Create(credentials: platformCredentials),
+                DeploymentServiceClient.Create(credentials: platformCredentials),
+                insecureConnection: false);
+
+//            Console.WriteLine("Starting Spatiald Scenario");
+//            var spatialdEndpoint = new PlatformApiEndpoint("localhost", SpatialdPort, true);
+//            RunScenario(
+//                PlayerIdentityTokenServiceClient.Create(spatialdEndpoint),
+//                LoginTokenServiceClient.Create(spatialdEndpoint),
+//                DeploymentServiceClient.Create(spatialdEndpoint),
+//                insecureConnection: true);
+        }
+
+        private static void RunScenario(string locatorHost, PlayerIdentityTokenServiceClient pitClient, LoginTokenServiceClient ltClient,
+            DeploymentServiceClient dsClient, bool insecureConnection)
+        {
+            var deployment = Setup(dsClient);
+            try
+            {
+                var pit = CreatePIT(pitClient);
+                deployment = FindDeployment(dsClient, deployment.Id);
+                var loginToken = CreateLoginTokenForDeployment(ltClient, pit, deployment.Id);
+                ConnectToLocator(locatorHost, insecureConnection, pit, loginToken);
+            }
+            finally
+            {
+                Cleanup(dsClient, deployment);
+            }
+        }
+
+        private static string CreatePIT(PlayerIdentityTokenServiceClient pitClient)
+        {
+            Console.WriteLine("Generate a PIT");
+            var playerIdentityTokenResponse = pitClient.CreatePlayerIdentityToken(
+                new CreatePlayerIdentityTokenRequest
+                {
+                    Provider = "provider",
+                    PlayerIdentifier = "player_identifier",
+                    ProjectName = ProjectName,
+                });
+            string pit = playerIdentityTokenResponse.PlayerIdentityToken;
```

Comment:
Do we need to check the error case in the scenario? Also you could just return the pit directly without making a variable

Reply:
niet. Fixed